### DISCUSSION TITLE
Default marker for geojson

### DISCUSSION THREAD

**Discussion by AndiLeni** [maplibre/maplibre-gl-js#4326] [Q&A]
Hello,

I followed these examples: https://maplibre.org/maplibre-gl-js/docs/examples/popup-on-click/ and https://maplibre.org/maplibre-gl-js/docs/examples/popup-on-hover.
Both of them show how to use a custom marker.
This may sound dumb, but how do I use the default marker in such a usecase which ships with maplibre?

If I have to use customized markers, ho may I change their size?

Thanks and kind regards

**Comment by HarelM** [comment]
Marker and popup are similar but a bit different. 
I would advise to checkout the API documentation, try out a few things until you get it right.
https://maplibre.org/maplibre-gl-js/docs/API/classes/Marker/
Also try to check with chatgpt or gimini etc.

**Reply by AndiLeni** [reply]
I already checked that page (and chatgpt 😄).
I still want the popups, they should be triggered by clicking on a marker. 
But I do not want a custom marker, I'd rather use a default marker which ships with maplibre if this is possible.

**Reply by HarelM** [reply]
There is a default marker and a default (empty) popup.

**Reply by geira** [reply]
Note that the question is about rendering **_GeoJSON_**, for which there doesn't seem to be any support for markers, let alone a default symbol.

Ideally, after using `map.addSource()` with `'type': 'geojson'` you should have been able to render points with a marker, but that isn't supported. Instead you can use `symbol`, but that doesn't have any default icon either so nothing is rendered unless you provide a symbol URL. In the end I chose to render the points as filled circles:

```
            map.addSource('nlaroutes', {
                'type': 'geojson',
                'data': 'https://api.met.no/weatherapi/nlaroutes/1.0/locations'
            });
            map.addLayer({
                'id': 'locations',
                'type': 'circle',
                'source': 'nlaroutes',
                'filter': ['==', "$type", "Point"],
                'paint': {
                    'circle-color': '#933',
                    'circle-radius': 5,
                    'circle-stroke-width': 1,
                    'circle-stroke-color': '#fff'
                }
            });
```
